rollup.base.config.js

```diff
@@ -0,0 +1,472 @@
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import { uglify } from "@microsoft/applicationinsights-rollup-plugin-uglify3-js";
+import replace from "@rollup/plugin-replace";
+import cleanup from "rollup-plugin-cleanup";
+import dynamicRemove from "@microsoft/dynamicproto-js/tools/rollup/node/removedynamic";
+import { es3Poly, es3Check, importCheck } from "@microsoft/applicationinsights-rollup-es3";
+
+const treeshakeCfg = {
+    moduleSideEffects: false,
+    propertyReadSideEffects: false,
+    tryCatchDeoptimization: false,
+    correctVarValueBeforeDeclaration: false
+};
+
+function doCleanup() {
+  return cleanup({
+    comments: [
+      'some', 
+      /^.\s*@DynamicProtoStub/i,
+      /^\*\*\s*@class\s*$/
+    ]
+  })
+}
+
+const browserRollupConfigFactory = (banner, importCheckNames, targetType, theNameSpace, entryInputName, outputName, libVersion, isProduction, format = 'umd', postfix = '', teamExt = '', replaceValues, treeshakeConfig) => {
+
+    var thePostfix = `${postfix}`;
+    if (libVersion) {
+        thePostfix = `.${libVersion}${postfix}`; 
+    }
+
+    var outputPath = `browser/${outputName}${teamExt}${thePostfix}.js`;
+    var prodOutputPath = `browser/${outputName}${teamExt}${thePostfix}.min.js`;
+    var inputPath = `dist-${targetType}/${entryInputName}.js`;
+
+    const browserRollupConfig = {
+        input: inputPath,
+        output: {
+            file: outputPath,
+            banner: banner,
+            format: format,
+            name: theNameSpace.browser,
+            extend: true,
+            freeze: false,
+            sourcemap: true,
+        },
+        treeshake: treeshakeConfig,
+        plugins: [
+            dynamicRemove(),
+            replace({
+                preventAssignment: true,
+                delimiters: ["", ""],
+                values: replaceValues
+            }),
+            importCheck({ exclude: importCheckNames }),
+            nodeResolve({
+                browser: false,
+                preferBuiltins: false
+            }),
+            doCleanup(),
+            es3Poly(),
+            es3Check()
+        ]
+    };
+
+
+    if (isProduction) {
+        browserRollupConfig.output.file = prodOutputPath;
+        browserRollupConfig.plugins.push(
+            uglify({
+                ie8: false,
```

Comment:
nit: I am just curious that do we set ie8 to false for this new change?

Reply:
oh right, that may cause a problem. Thanks for pointing that out. Let's wait for Nev's suggestion.